python/MOCK_sensors.py

```diff
@@ -122,8 +130,148 @@ def publish(self):
 
         if rospy.get_param('sensor_noise', default=False):
             data = self.add_noise(data)
+
+        brokenWindSensorStatus = rospy.get_param('broken_wind_sensors', default="not_broken")
+        if brokenWindSensorStatus != "not_broken":
+            rospy.logwarn_throttle(30, "Broken Wind Sensor Status: " + brokenWindSensorStatus)
+            self.breakWindSensors(data, brokenWindSensorStatus)
+
         self.publisher.publish(data)
 
+    def breakWindSensors(self, data, brokenWindSensorStatus):
+        # Corrupts wind sensor data depending on the setting specified by string brokenSensorStatus
+
+        if (brokenWindSensorStatus == "stuck_at_zero"):
+            # This setting chooses a wind sensor which reads an angle of 0 and speed 0
+
+            if(self.brokenSensorData is None):
+                randomSeed = get_rosparam_or_default_if_invalid('random_seed', default=None, rosparam_type_cast=str)
+                random.seed(randomSeed)
+                self.brokenSensorData = random.randint(1, 3)
```

Comment:
Since these lines are executed in every branch, they can be moved to before the if-else statements

Reply:
Instead of directly setting `self.brokenSensorData` when moved before the if-else statements, can create a new variable (e.g. `brokenSensorID`)